pkg/cluster/kubeclient/backup.go

```diff
@@ -29,10 +30,18 @@ func (u *Kubeclientset) BackupCluster(ctx context.Context, backupName string) er
 	}
 
 	job.Spec.BackoffLimit = to.Int32Ptr(0)
-	job.Spec.Template.Spec.Containers[0].Args = []string{
-		"etcdbackup",
-		fmt.Sprintf("--blobName=%s", backupName),
-		"--action=save",
+
+	// Special code to make etcd backup work for v4+ branches
+	// if image is etcdbackup - old cluster
+	// else - new
+	if strings.Contains(cronjob.Spec.JobTemplate.Spec.Template.Spec.Containers[0].Image, "etcdbackup") {
```

Comment:
sorry, what does this change do and why do we need it in 7.1?

Reply:
v7 was never designed to run updated on v4. with image consolidation v7 branch way of initializing etcbackup does not work on v4 image.

Reply:
ah, I see - and because we run a backup on the cluster before we upgrade it, we need this to work. sgtm.